notes/connection-lifecycle.md

```diff
@@ -0,0 +1,83 @@
+# Connection lifecycle
+
+## Current implementation
```

Comment:
Should we update those notes so that they remain true after we merge the pull request? They still mention `set_cert`, `auto_open`, etc.

Reply:
Done in https://github.com/urllib3/urllib3/pull/2768/commits/f25fec70eb7f4661ad8f4692fc6d8997413c122d